phdi/fhir/transport/http.py

```diff
@@ -76,47 +79,54 @@ def upload_bundle_to_fhir_server(
+    :param max_bundle_size: The maximum number of resources per bundle to upload to
+      the FHIR server.
+    :return: A `requests.Response` object containing the response from the FHIR server.
     """
 
     access_token = cred_manager.get_access_token()
-
-    response = http_request_with_reauth(
-        cred_manager=cred_manager,
-        url=fhir_url,
-        retry_count=3,
-        request_type="POST",
-        allowed_methods=["POST"],
-        headers={
-            "Authorization": f"Bearer {access_token}",
-            "Accept": "application/fhir+json",
-            "Content-Type": "application/fhir+json",
-        },
-        data=bundle,
-    )
-
-    # FHIR uploads are sent as a batch.  Although the batch succeeds,
-    # individual entries within the batch may fail, so we log them here
-    if response.status_code == 200:
-        response_json = response.json()
-        entries = response_json.get("entry", [])
-        for entry_index, entry in enumerate(entries):
-            entry_response = entry.get("response", {})
-
-            # FHIR bundle.entry.response.status is string type - integer status code
-            # plus may inlude a message
-            if entry_response and not entry_response.get("status", "").startswith(
-                "200"
-            ):
-                _log_fhir_server_error(
-                    status_code=int(entry_response["status"][0:3]),
-                    batch_entry_index=entry_index,
-                )
-
-    else:
-        _log_fhir_server_error(response.status_code)
-
-    return response
+    responses = []
+
+    # ensure that bundles are below the set maximum size of resources
+    split_bundles = _split_bundle_resources(bundle, max_bundle_size)
+
+    for single_bundle in split_bundles:
+        response = http_request_with_reauth(
+            cred_manager=cred_manager,
+            url=fhir_url,
+            retry_count=3,
+            request_type="POST",
+            allowed_methods=["POST"],
+            headers={
+                "Authorization": f"Bearer {access_token}",
+                "Accept": "application/fhir+json",
+                "Content-Type": "application/fhir+json",
+            },
+            data=single_bundle,
+        )
+        # FHIR uploads are sent as a batch.  Although the batch succeeds,
+        # individual entries within the batch may fail, so we log them here
+        if response.status_code == 200:
+            response_json = response.json()
+
+            entries = response_json.get("entry", [])
+            for entry_index, entry in enumerate(entries):
+                entry_response = entry.get("response", {})
+
+                # FHIR bundle.entry.response.status is string type - integer status code
+                # plus may inlude a message
+                if entry_response and not entry_response.get("status", "").startswith(
+                    "200"
+                ):
```

Comment:
I think this will make us consistent with what we do in the container an handle 201s. We have seen differences across FHIR servers and differences when a resource already exists v. when it is created for the first time.

```suggestion
                if entry_response and entry_response.get("status", "") not in ["200 OK", "201 Created", "200", "201"]:
```